web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/TextSearchFilterServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+        if (textSearch == null || textSearch.isEmpty()) {
+            return sickLeaves;
+        }
+        return filterOnTextSearch(sickLeaves, textSearch);
+    }
+
+    @Override
+    public boolean filter(SjukfallEnhet sickLeave, String textSearch) {
+        if (textSearch == null || textSearch.isEmpty()) {
+            return true;
+        }
+        return textSearchMatchesAnyField(sickLeave, textSearch);
+    }
+
+    private List<SjukfallEnhet> filterOnTextSearch(List<SjukfallEnhet> sickLeaves, String textSearch) {
+        return sickLeaves.stream()
+            .filter(sickLeave -> textSearchMatchesAnyField(sickLeave, textSearch))
+            .collect(Collectors.toList());
+    }
+
+    private boolean textSearchMatchesAnyField(SjukfallEnhet sickLeave, String textSearch) {
+        return searchMatchesAge(sickLeave, textSearch)
+            || searchMatchesPatientName(sickLeave, textSearch)
+            || searchMatchesPatientGender(sickLeave, textSearch)
+            || searchMatchesDiagnosisCode(sickLeave, textSearch)
+            || searchMatchesSickLeavePeriod(sickLeave, textSearch) || searchMatchesLength(sickLeave, textSearch)
+            || searchMatchesNumberOfCertificates(sickLeave, textSearch)
+            || searchMatchesActiveDegree(sickLeave, textSearch)
+            || searchMatchesDoctorName(sickLeave, textSearch)
+            || searchMatchesPatientId(sickLeave, textSearch);
+    }
+
+    private boolean searchMatchesPatientId(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getPatient().getId(), textSearch);
+    }
+
+    private boolean searchMatchesDoctorName(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getLakare().getNamn(), textSearch);
+    }
+
+    private boolean searchMatchesActiveDegree(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getAktivGrad() + "%", textSearch);
+    }
+
+    private boolean searchMatchesNumberOfCertificates(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(String.valueOf(sickLeave.getIntygLista().size()), textSearch);
+    }
+
+    private boolean searchMatchesLength(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(String.valueOf(sickLeave.getDagar()), textSearch);
+    }
+
+    private boolean searchMatchesSickLeavePeriod(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(sickLeave.getStart().toString(), textSearch) || matches(sickLeave.getSlut().toString(), textSearch);
+    }
+
+    private boolean searchMatchesDiagnosisCode(SjukfallEnhet sickLeave, String textSearch) {
+        return matches(getDiagnosis(sickLeave), textSearch);
+    }
+
+    private static String getDiagnosis(SjukfallEnhet sickLeave) {
+        final var diagnosis = new StringBuilder(sickLeave.getDiagnosKod().getName() + sickLeave.getDiagnosKod().getCleanedCode());
```

Comment:
Maybe you should consider building the string for diagnosis in a similar way as the client does it... at least consider adding the code before the name.

Reply:
Fixed it!